tutorials/add_model.md

```diff
@@ -0,0 +1,139 @@
+
+        item_id: int, optional, default: None
+            The index of the item for that to perform score prediction.
+            If None, scores for all known items will be returned.
+
+        Returns
+        -------
+        res : A scalar or a Numpy array
+            Relative scores that the user gives to the item or to all known items
+        """
+``` 
+Every model is an extension of the `Recommender` class. All you need to do is to redefine the functions listed in the above `recom_pmf.py` file.
+
+The `fit()` function should contain or make a call to the necessary codes to fit your model to the training data, while the `score()` function specifies how to perform predictions using your model.
+ 
+#### More on the score function 
+When you perform rating prediction evaluation (e.g., in terms of RMSE) Cornac calls the `Recommender.rate()` function, which in turn calls the `score()` function and clips the predicted values to lie in the same range as the original ratings. Analogously, Cornac calls the `Recommender.rank()` function, which returns a ranking of items according to their scores as predicted by the function `score()`. We recommend you to take a look at the `rate()` and `rank()` functions inside the `Recommender` class, as you may wish to redefine them as well, if the default settings do not meet your needs. 
+
+### 2. Making your recommender available to Cornac
+As you may have already noticed, Cornac treats each recommender model as a separate module, so as to reduce coupling and ease the contribution of new models. This is the reason why you need to include a `pmf/__init__.py` file:
+```python
+from .recom_pmf import PMF
+```
+
+One last step is required to make you recommender model available to Cornac, which is to update `../models/__init__.py` by adding the following line:
+```python
+from .pmf import PMF
+```
+
+### 3. Indicating dependencies
+The file `requirements.txt` is optional as opposed to the others, and it is only needed if your implementation relies on some external libraries/frameworks, e.g., TensorFlow, PyTorch, etc., in which case you should include this file and indicate which versions of the framework(s) are required. Here is a sample of a `requirements.txt` file:
+```
+tensorflow>=1.10.0
+```
+
+### 4. Adding documentation
+
+Cornac uses [Sphinx](http://www.sphinx-doc.org/en/master/) to generate documentation. All you need to do is to fulfill the Parameters section inside `recom_pmf.py`. If you are not familiar with Sphinx syntax conventions, you can just take a look at some of the existing Cornac models and follow them.
+
+Once you have documented the parameters of your recommender, the next step, to integrate your model to the docs, is to update `cornac/docs/source/models.rst` by adding the following code:
+```
+Probabilitic Matrix Factorization (PMF)
+-----------------------------------------
+.. automodule:: cornac.models.pmf.recom_pmf
+   :members:
+```
+
```

Comment:
In the `README.md`, there is a table listing all the available models in Cornac covering models' name, paper/reference, additional dependencies, and examples. You are also expected to update the table for showing the availability of your model to other users.